### PR TITLE
feat: Add version page

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Screen Shot 2022-11-14 at 13 52 32

Ps: Adding tests in the next commit - I forgot 🤦

### CODE REVIEW

File: site/src/pages/TemplateVersionPage/TemplateVersionPage.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+              <StatsItem
+                label="Template"
+                value={<Link to={`/templates/${template}`}>{template}</Link>}
+              />
+              <StatsItem
+                label="Created by"
+                value={version.created_by.username}
+              />
+              <StatsItem
+                label="Created at"
```
Comment (presleyp) [inline site/src/pages/TemplateVersionPage/TemplateVersionPage.tsx:84]: Maybe "Created" reads better

File: site/js-untar.d.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+declare module "js-untar" {
```
Comment (presleyp) [inline site/js-untar.d.ts:1]: Does js-untar not come with a type definition file, so you had to write one?

File: site/src/AppRouter.tsx [hunk 3]
```diff
@@ -123,16 +126,14 @@ export const AppRouter: FC = () => {
...
-          />
-        </Route>
+        <Route
+          path="workspaces"
+          element={
+            <AuthAndFrame>
+              <WorkspacesPage />
+            </AuthAndFrame>
+          }
+        />
```
Comment (presleyp) [inline site/src/AppRouter.tsx:136]: What does this change do?

File: site/src/components/TemplateStats/TemplateStats.tsx [hunk 4]
```diff
@@ -26,77 +27,39 @@ export const TemplateStats: FC<TemplateStatsProps> = ({
...
-        <span className={styles.statsValue}>{template.created_by_name}</span>
-      </div>
-    </div>
+    <Stats>
+      <StatsItem
+        label={Language.usedByLabel}
+        value={
+          <>
+            {formatTemplateActiveDevelopers(template.active_user_count)}{" "}
+            {template.active_user_count === 1
```
Comment (presleyp) [inline site/src/components/TemplateStats/TemplateStats.tsx:37]: If you feel like messing with it, i18n has a way of handling this

File: site/src/pages/TemplateVersionPage/TemplateVersionPage.tsx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+      <Margins>
+        <PageHeader>
+          <PageHeaderCaption>Versions</PageHeaderCaption>
+          <PageHeaderTitle>{versionName}</PageHeaderTitle>
+        </PageHeader>
+      </Margins>
+
+      {!files && !state.matches("done.error") && <Loader />}
+
+      {state.matches("done.ok") && version && files && (
```
Comment (presleyp) [inline site/src/pages/TemplateVersionPage/TemplateVersionPage.tsx:71]: looks like there are some cases where neither the loader nor the data will render. Should it render an error or anything in that case?

File: site/js-untar.d.ts (hunk 2 above)
Comment (BrunoQuaresma) [reply site/js-untar.d.ts:1]: Exactly

File: site/src/AppRouter.tsx (hunk 3 above)
Comment (BrunoQuaresma) [reply site/src/AppRouter.tsx:136]: Nothing, just simplify things. I was chasing a bug and I noticed this.

File: site/src/components/TemplateStats/TemplateStats.tsx (hunk 4 above)
Comment (BrunoQuaresma) [reply site/src/components/TemplateStats/TemplateStats.tsx:37]: I will take a look. I have to move all the language stuff to i18n tho

File: site/src/pages/TemplateVersionPage/TemplateVersionPage.tsx (hunk 5 above)
Comment (BrunoQuaresma) [reply site/src/pages/TemplateVersionPage/TemplateVersionPage.tsx:71]: Yes!

File: site/src/components/TemplateStats/TemplateStats.tsx (hunk 4 above)
Comment (presleyp) [reply site/src/components/TemplateStats/TemplateStats.tsx:37]: No worries if you want to put it off. I think there's an example in the `common` file.

File: site/src/xServices/templateVersion/templateVersionXService.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            target: "done.ok",
+            actions: ["assignFiles"],
+          },
+          onError: {
+            target: "done.error",
+            actions: ["assignError"],
+          },
+        },
+      },
+      done: {
```
Comment (presleyp) [inline site/src/xServices/templateVersion/templateVersionXService.ts:66]: might be simpler to have "ok" and "error" as top-level states?

File: site/src/xServices/templateVersion/templateVersionXService.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+          ok: { type: "final" },
+          error: { type: "final" },
+        },
+      },
+    },
+  },
+  {
+    actions: {
+      assignError: assign({
+        error: (_, { data }) => data,
```
Comment (presleyp) [inline site/src/xServices/templateVersion/templateVersionXService.ts:77]: nice simple way to just show one error at a time (which is probably all that's possible in this case, but maybe we should do it more often)

**Review by presleyp — APPROVED** [review]
With tests and error handling, should be good!

**Comment by Emyrk** [comment]
I assume all users cannot read the tf file though right?

**Comment by BrunoQuaresma** [comment]
> I assume all users cannot read the tf file though right?

Yes, I think so. So probably we want to protect the files route, right?

**Pull request merged**